src/ai/backend/common/task_runner/types.py

```diff
@@ -0,0 +1,136 @@
+import asyncio
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Optional
+
+from ai.backend.logging.utils import BraceStyleAdapter
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))
+
+
+class AbstractTask(ABC):
+    @classmethod
+    @abstractmethod
+    def name(cls) -> str:
+        raise NotImplementedError
+
+    @classmethod
+    @abstractmethod
+    def timeout(cls) -> Optional[float]:
+        """
+        Returns the timeout for the task
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def run(self) -> None:
+        raise NotImplementedError
+
+
+@dataclass
+class TaskRunnerArgs:
+    task: AbstractTask
+    interval: float
+    grace_period: float = 0
+    timeout: Optional[float] = None
```

Comment:
It seems unnecessary to receive `timeout` as an argument if it's already provided as an `abstractmethod` in the `AbstractTask`.